packages/foundry-sdk-generator/src/generate/betaClient/minifyBundle.ts

```diff
@@ -150,16 +223,47 @@ export class ProjectMinifier {
               const importsFromNamespace = new Set(
                 exportSourceFile.getExportedDeclarations().keys(),
               );
-              this.pushNextVisit(exportSourceFile, importsFromNamespace);
+              this.pushNextVisit(
+                exportSourceFile,
+                importsFromNamespace,
+                "minifyProjectA",
+              );
             }
           }
         }
 
         const namedExports = declaration.getNamedExports();
 
+        function deleteFromSet(set: Set<string>, toDelete: Set<string>) {
+          for (const d of toDelete) {
+            set.delete(d);
+          }
+        }
+
         // export * from "..."
         if (namedExports.length === 0) {
-          this.pushNextVisit(exportSourceFile, visitImports.imports);
+          const depExportNames = new Set(
+            exportSourceFile.getExportSymbols().map(s => s.getName()).filter(
+              s => visitImports.imports.has(s),
+            ),
+          );
+          if (DEBUG) {
+            // eslint-disable-next-line no-console
+            console.log(
+              chalk.green(
+                `${logNode(declaration)} removing ${
+                  chalk.white([...depExportNames].join(", "))
+                } from import set of ${logFilename(sourceFile)}`,
+              ),
+            );
+          }
+
+          deleteFromSet(visitImports.imports, depExportNames);
```

Comment:
For each thing that the `"otherFile"` had in our list to include, we need to remove it from our current work list so that subsequent `export *` do not over include.